packages/integration-tests/test/deposit.spec.ts

```diff
@@ -0,0 +1,129 @@
+  }
+}
+
+describe('Transactions', () => {
+  let l1Provider: JsonRpcProvider
+  let l1Wallet: Wallet
+  let l2Provider: JsonRpcProvider
+  let addressResolver: Contract
+  let postgres: PostgresDB
+
+  before(async () => {
+    postgres = new PostgresDB('postgres', 5432, 'test', 'test', 'rollup')
+    l1Provider = new JsonRpcProvider(Config.L1NodeUrlWithPort())
+    l1Wallet = Wallet.fromMnemonic(mnemonic).connect(l1Provider)
+    const web3 = new Web3Provider(
+      ganache.provider({
+        mnemonic,
+      })
+    )
+
+    l2Provider = new OptimismProvider(Config.L2NodeUrlWithPort(), web3)
+
+    const key = add0x(Config.l1ContractDeploymentPrivateKey())
+    // Set up address resolver which we can use to resolve any required contract addresses
+    const deployerAddress = computeAddress(key)
+    const addressResolverAddress = getContractAddress({
+      from: deployerAddress,
+      nonce: 0,
+    })
+    const AddressResolverFactory = getContractFactory('AddressResolver')
+    addressResolver = AddressResolverFactory.connect(l1Wallet).attach(
+      addressResolverAddress
+    )
+    // Depends on custom fork of ganache
+    // turn on automine
+    await l1Provider.send('evm_mine_interval', [2])
+  })
+
+  after(async () => {
+    // turn off automine
+    await l1Provider.send('evm_mine_interval', [0])
+  })
+
+  it('should send a lot of transactions', async () => {
+    // Set up L1ToL2TransactionQueue contract object
+    const l1ToL2TransactionQueueAddress = await addressResolver.getAddress(
+      'L1ToL2TransactionQueue'
+    )
+    const L1ToL2TransactionQueueFactory = getContractFactory(
+      'L1ToL2TransactionQueue'
+    )
+    const l1ToL2TransactionQueue: Contract = L1ToL2TransactionQueueFactory.connect(
+      l1Wallet
+    ).attach(l1ToL2TransactionQueueAddress)
+
+    // Send an L1ToL2Transaction
+    const input = ['0x' + '01'.repeat(20), 500_000, '0x' + '00']
+    const calldata = await l1ToL2TransactionQueue.interface.encodeFunctionData(
+      'enqueueL1ToL2Message',
+      input
+    )
+
+    const txResponse = await l1Wallet.sendTransaction({
+      data: calldata,
+      to: l1ToL2TransactionQueueAddress,
+    })
+    const txReceipt = await txResponse.wait()
+
+    const getL1RollupTx = () =>
+      postgres.select(
+        `SELECT * FROM l1_rollup_tx WHERE l1_tx_hash = '${txResponse.hash}'`
+      )
+    const l1RollupTxs = await poll(getL1RollupTx, 10_000)
+    const l1RollupTx = l1RollupTxs[0]
+
+    const address = await l1Wallet.getAddress()
+    l1RollupTx.l1_message_sender.should.equal(address)
+    l1RollupTx.queue_origin.should.equal(QueueOrigin.L1_TO_L2_QUEUE)
+
+    const getQueueResult = () =>
+      postgres.select(
+        `SELECT * FROM geth_submission_queue WHERE l1_tx_hash = '${txResponse.hash}'`
+      )
+    const queueResults = await poll(getQueueResult, 10_000)
+    const queueResult = queueResults[0]
+    queueResult.l1_tx_hash.should.equal(txResponse.hash)
+  }).timeout(1000000)
```

Comment:
This is still probably a pretty intense timeout.

One note that we learned from @K-Ho is just that we need to add tooling for deploying the L2Messenger contract automatically & then exposes the addresses. ( just threw together a ticket https://github.com/ethereum-optimism/roadmap/issues/79 ) 

The reason I bring this up here is maybe we should add a `// TODO` or something like that to test our deposit against the L2Messenger (verify that it got include), and maybe even use the watcher service eventually. The reason this is top of mind is our current integration tests didn't catch the error we had before with the `l1MsgSender`

Reply:
I lowered the timeout